backend/schema/visit.go

```diff
@@ -15,11 +15,16 @@ func Visit(n Node, visit func(n Node, next func() error) error) error {
 // VisitExcludingMetadataChildren visits all nodes in the schema except the children of metadata nodes.
 // This is used when generating external modules to avoid adding imports only referenced in the bodies of
 // stubbed verbs.
-func VisitExcludingMetadataChildren(n Node, visit func(n Node, next func() error) error) error {
-	return visit(n, func() error {
+func VisitExcludingMetadataChildren(n Node, visit func(n Node, parents []Node, next func() error) error) error {
+	return innerVisitExcludingMetadataChildren(n, []Node{}, visit)
+}
+
+func innerVisitExcludingMetadataChildren(n Node, parents []Node, visit func(n Node, parents []Node, next func() error) error) error {
```

Comment:
I think it would be simpler to type switch on `Verb`, manually call `Visit()` on the Verb's children iff they're not Unit and don't call `next()` in that case.

Reply:
Isn't that an odd separation of concerns though? As in the needs of a specific go scaffolding use case compared to `schema/visit.go` providing ways to traverse the graph.
Currently this is also called when scaffolding Kotlin (which to be fair, might not be affected by a change like that, not 100% sure).

Reply:
Ah I see what you're saying, but no - this is supposed to be universal. It should be true in Kotlin too.